src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpUploader.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.slc.services.steps.sftpupload;
+
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.sftp.SFTPClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.PdfDoc;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.exceptions.FtpStepException;
+
+import java.io.IOException;
+
+@Component
+public class FtpUploader {
+
+    private static final Logger logger = LoggerFactory.getLogger(FtpUploader.class);
+
+    private final String hostname;
+    private final int port;
+    private final String fingerprint;
+    private final String username;
+    private final String password;
+    private final SSHClient ssh;
+
+    // region constructor
+    public FtpUploader(
+        @Value("${sftp.hostname}") String hostname,
+        @Value("${sftp.port}") int port,
+        @Value("${sftp.fingerprint}") String fingerprint,
+        @Value("${sftp.username}") String username,
+        @Value("${sftp.password}") String password,
+        SSHClient sshClient
+    ) {
+        this.hostname = hostname;
+        this.port = port;
+        this.fingerprint = fingerprint;
+        this.username = username;
+        this.password = password;
+        this.ssh = sshClient;
+    }
+    // endregion
+
+    public void upload(PdfDoc pdfDoc) {
+        try {
+
+            ssh.addHostKeyVerifier(fingerprint);
+            ssh.connect(hostname, port);
+            ssh.authPassword(username, password);
+
+            try (SFTPClient sftp = ssh.newSFTPClient()) {
```

Comment:
try with no catch block?.Is this try required as there is try at the start?

Reply:
this is a 'try-with-resource'

Reply:
👍